File: apps/news-spider-cave/src/core/vo/CommonResult.ts
export class CommonResult<T> {
    static CODE_SUCCESS = 0;
    code: number;
    message: string;
    data: T;

    constructor(code: number, message: string, data: T) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    static success<T>(data: T): CommonResult<T> {
        return new CommonResult(CommonResult.CODE_SUCCESS, "success", data);
    }

    static error<T>(code: number, message: string): CommonResult<T> {
        return new CommonResult(code, message, null);
    }
}

File: apps/servant/src/main/java/top/xrepo/servant/core/vo/CommonResult.java
package top.xrepo.servant.core.vo;

import java.io.Serializable;

public class CommonResult <T> implements Serializable {
    public static Integer CODE_SUCCESS = 0;

    /**
     * 错误码
     */
    private Integer code;
    /**
     * 错误信息
     */
    private String message;
    /**
     * 数据
     */
    private T data;

    public CommonResult(Integer code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    public static <T> CommonResult<T> success(T data) {
        return new CommonResult<>(CODE_SUCCESS, "成功", data);
    }

    public static <T> CommonResult<T> error(Integer code, String message) {
        return new CommonResult<>(code, message, null);
    }

    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "CommonResult{" +
                "code=" + code +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }
}
